src/core_ocean/shared/mpas_ocn_diagnostics.F

```diff
@@ -1496,12 +1498,15 @@ subroutine ocn_compute_KPP_input_fields(statePool, forcingPool, meshPool, diagno
        ! transport code.  This includes tracer forcing due to thickness
 
        nonLocalSurfaceTracerFlux(indexTempFlux, iCell) = activeTracersSurfaceFlux(indexTempFlux,iCell) &
-               + penetrativeTemperatureFlux(iCell) - penetrativeTemperatureFluxOBL(iCell) - fracAbsorbed * &
-               surfaceThicknessFlux(iCell) * activeTracers(indexTempFlux,1,iCell) + &
-               activeTracersSurfaceFluxRunoff(indexTempFlux,iCell) * fracAbsorbedRunoff
+               + penetrativeTemperatureFlux(iCell) - penetrativeTemperatureFluxOBL(iCell)  &
+               - fracAbsorbed * (rainFlux(iCell) + evaporationFlux(iCell)) * activeTracers(indexTempFlux,1,iCell)/rho_sw  &
+               - fracAbsorbedRunoff * surfaceThicknessFluxRunoff(iCell) * (  &
+                   max(activeTracers(indexTempFlux,1,iCell),0.0_RKIND) &
+                 + min(activeTracers(indexTempFlux,1,iCell),0.0_RKIND)    )/rho_sw
```

Comment:
@maltrud Please check this:
```
  max(activeTracers(indexTempFlux,1,iCell),0.0_RKIND) &
+ min(activeTracers(indexTempFlux,1,iCell),0.0_RKIND) 
```
is equal to:
```
activeTracers(indexTempFlux,1,iCell)
```
so that must not be what you meant.

Reply:
@mark-petersen , i actually did this on purpose to denote that there are 2 distinct cases depending on whether T>0 or T<0.  if T>0, then we need to subtract off the runoff surface T flux because it doesn't change the surface buoyancy.  if T<0, then no "effective flux" was needed, but it does change buoyancy so we need to add it in.

but you're right, it is confusing and should probably be in a comment instead.

Reply:
@mark-petersen , @vanroekel discovered an error.  the line with the 'max' needs to be removed:

max(activeTracers(indexTempFlux,1,iCell),0.0_RKIND)
then the code should be ready.  can you modify the code, or should i?

Reply:
I just did it. It compiles. Check the revised code - is that what you want?